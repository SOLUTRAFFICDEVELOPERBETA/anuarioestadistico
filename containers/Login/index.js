import React from 'react';
import EyeIcon from 'mdi-react/EyeIcon';
import KeyVariantIcon from 'mdi-react/KeyVariantIcon';
import AccountOutlineIcon from 'mdi-react/AccountOutlineIcon';
import styled from '@emotion/styled';
import AlertContext from '../../contexts/alert';
import AuthContext from '../../contexts/auth';
import Link from 'next/link';
import { css } from '@emotion/core';
import { ButtonGroup, Button } from '@material-ui/core';
import { useRouter } from 'next/router';

// Estilos del Formulario
const Form = styled.form`
    display: flex;
    flex-wrap: wrap;

    .signIn {
        padding: 15px;
        background-color: #0d283d;
        border-color: #0d283d;
        width: 100%;
        border-radius: 5px;
        margin-bottom: 15px;
        font-size: 14px;
        overflow: hidden;
        color: #fff;
        text-align: center;
        border: 1px solid;
    }
    .createAccount {
        text-decoration: none;
        cursor: pointer;
        color: #0d283d;
        border-color: #0d283d;
        border: 1px solid;
        width: 100%;
        border-radius: 5px;
        margin-bottom: 20px;
        font-size: 14px;
        overflow: hidden;
        text-align: center;
    }
`;

const AccountCard = styled.div`
    background-color: #ffffff;
    padding: 10px 40px;
    max-width: 520px;
    width: 100%;
    border-radius: 10px;
`;

const AccountHead = styled.div`
    margin-bottom: 30px;
    padding-left: 10px;
    border-left: 4px solid #0d283d;

    h3 {
        color: #646777;
        font-size: 24px;
        line-height: 32px;

        .accountlogo {
            font-weight: 700;
            padding-left: 5px;
        }
        .accountlogoaccent {
            color: #0d283d;
        }
    }
    h4 {
        font-size: 12px;
        color: #999999;
        line-height: 16px;
    }
`;
// Estilos del Input
const FormGroup = styled.div`
    width: 100%;
    position: relative;

    span {
        color: #646777;
        display: inline-block;
    }

    .formformgroupfield {
        width: 100%;
        display: flex;
        margin: auto;
    }
    .formformgroupicon {
        background: #f2f4f7;
        border: 1px solid #f2f4f7;
        padding: 6px;
        height: 32px;
    }
    input {
        width: 100%;
        padding: 5px 10px;
        font-size: 12px;
        height: 42px;
        transition: border 0.3s;
        background: transparent;
        border: 1px solid #f2f4f7;
        color: #646777;
    }
    .formformgrouplabel {
        color: #646777;
        display: inline-block;
    }
    button {
        background: #f2f4f7;
        border: 1px solid #f2f4f7;
        cursor: pointer;
        padding: 4px;
        height: 32px;
    }
`;

/**
 * @param Componente que permite el inicio de sesión al usuario.
 * @see Link
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 * consulte los estilos del componente en {@link https://emotion.sh/docs/styled | Emotion}
 */
const LogInForm = () => {
    const { showMessage } = React.useContext(AlertContext);
    const { onLogIn } = React.useContext(AuthContext);
    const [showPassword, setShowPassword] = React.useState(false);
    const [data, setData] = React.useState({
        password: '',
        email: ''
    });
    const Router = useRouter()
    // Método para mostrar la contraseña.
    const fshowPassword = (e) => {
        e.preventDefault();
        setShowPassword(!showPassword);
    };

    const handlenSubmit = (e) => {
        e.preventDefault();
        const { password, email } = data;

        if (email.trim() === '' && password.trim() === '') {
            showMessage('Todos los campos son obligatorios', 'warning');
            return;
        }

        onLogIn(email, password);
    };

    const handlenOnchange = (e) => {
        const { value, name } = e.target;
        setData({ ...data, [name]: value });
    };

    return (
        <AccountCard>
            <AccountHead>
                <h3>
                    Bienvenido a<span className="accountlogo">Anuario-</span>
                    <span className="accountlogoaccent">Estadístico</span>
                </h3>
                <h4>Región Ordenada, Conectada y Sostenible</h4>
            </AccountHead>

            <Form onSubmit={handlenSubmit}>
                <FormGroup>
                    <span>Nombre del usuario</span>
                    <div className="formformgroupfield">
                        <div className="formformgroupicon">
                            <AccountOutlineIcon />
                        </div>
                        <input
                            onChange={handlenOnchange}
                            name="email"
                            type="text"
                            placeholder="Correo..."
                        />
                    </div>
                </FormGroup>
                <FormGroup>
                    <span className="formformgrouplabel">Contraseña</span>
                    <div className="formformgroupfield">
                        <div className="formformgroupicon">
                            <KeyVariantIcon />
                        </div>
                        <input
                            onChange={handlenOnchange}
                            name="password"
                            type={showPassword ? 'text' : 'password'}
                            placeholder="Contraseña..."
                        />
                        <button
                            className="formgroupbutton"
                            onClick={(e) => fshowPassword(e)}
                            type="button">
                            <EyeIcon />
                        </button>
                    </div>
                    <Link
                        href={'/auth/recoverPassword'}
                        css={css`
                            margin: 1rem;
                            font-size: 15px;
                            line-height: 15px;
                        `}>
                        <a
                            css={css`
                                text-decoration: none;
                                color: #0d283d;
                            `}>
                            Olvidaste tu contraseña?
                        </a>
                    </Link>
                </FormGroup>
                <ButtonGroup

                    orientation="vertical"
                    size="large"
                    fullWidth
                    style={{ margin: '10px 0' }}>
                    <Button
                        color="primary"
                        type="submit"
                        variant="contained">
                        Iniciar Sesión
                </Button>
                    {/* <Button
                        onClick={() => Router.push('/auth/register')}
                        variant="text">
                        Registrarme
                </Button> */}
                </ButtonGroup>

            </Form>
        </AccountCard>
    );
};

export default LogInForm;
