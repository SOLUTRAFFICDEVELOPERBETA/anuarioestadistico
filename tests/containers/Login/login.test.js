import '@testing-library/jest-dom';
import { act } from '@testing-library/react';
import { mount, shallow } from 'enzyme';
import React from 'react';
import Login from '../../../containers/Login';
import AlertContext from '../../../contexts/alert';
import AuthContext from '../../../contexts/auth';

describe('Pruebas en <Login/>', () => {
    const onLogIn = jest.fn();
    const showMessage = jest.fn();
    const wrapper = mount(
        <AlertContext.Provider value={{
            showMessage
        }}>
            <AuthContext.Provider value={{
                onLogIn
            }}>
                <Login />
            </AuthContext.Provider>
        </AlertContext.Provider>
    )

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();

    })

    test('no debe llamar la función onLogIn()', () => {
        wrapper.find('form').prop('onSubmit')({ preventDefault() { } })
        expect(onLogIn).not.toHaveBeenCalled();


    })

    test('debe llamar la función onLogIn()', () => {
        const values = {
            email: 'prueba@gmail.com',
            password: '12345678'
        }
        act(() => {
            wrapper.find('input').at(0).prop('onChange')({ target: { value: values.email, name: 'email' } })
            wrapper.find('input').at(1).prop('onChange')({ target: { value: values.password, name: 'password' } })
        })
        wrapper.find('form').prop('onSubmit')({ preventDefault() { } });
        expect(onLogIn).not.toHaveBeenCalled();


    })
})