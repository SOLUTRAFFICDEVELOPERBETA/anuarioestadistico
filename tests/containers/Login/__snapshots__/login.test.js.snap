// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Login/> debe mostrarse correctamente 1`] = `
<div
  className="css-1m18txs-AccountCard e6b47nl1"
>
  <div
    className="css-1pw56y9-AccountHead e6b47nl2"
  >
    <h3>
      Bienvenido a
      <span
        className="accountlogo"
      >
        Anuario-
      </span>
      <span
        className="accountlogoaccent"
      >
        Estadístico
      </span>
    </h3>
    <h4>
      Región Ordenada, Conectada y Sostenible
    </h4>
  </div>
  <form
    className="css-1tpjg3z-Form e6b47nl0"
    onSubmit={[Function]}
  >
    <div
      className="css-1my2ixp-FormGroup e6b47nl3"
    >
      <span>
        Nombre del usuario
      </span>
      <div
        className="formformgroupfield"
      >
        <div
          className="formformgroupicon"
        >
          <svg
            className="mdi-icon "
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,6A2,2 0 0,0 10,8A2,2 0 0,0 12,10A2,2 0 0,0 14,8A2,2 0 0,0 12,6M12,13C14.67,13 20,14.33 20,17V20H4V17C4,14.33 9.33,13 12,13M12,14.9C9.03,14.9 5.9,16.36 5.9,17V18.1H18.1V17C18.1,16.36 14.97,14.9 12,14.9Z"
            />
          </svg>
        </div>
        <input
          name="email"
          onChange={[Function]}
          placeholder="Correo..."
          type="text"
        />
      </div>
    </div>
    <div
      className="css-1my2ixp-FormGroup e6b47nl3"
    >
      <span
        className="formformgrouplabel"
      >
        Contraseña
      </span>
      <div
        className="formformgroupfield"
      >
        <div
          className="formformgroupicon"
        >
          <svg
            className="mdi-icon "
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M22,18V22H18V19H15V16H12L9.74,13.74C9.19,13.91 8.61,14 8,14A6,6 0 0,1 2,8A6,6 0 0,1 8,2A6,6 0 0,1 14,8C14,8.61 13.91,9.19 13.74,9.74L22,18M7,5A2,2 0 0,0 5,7A2,2 0 0,0 7,9A2,2 0 0,0 9,7A2,2 0 0,0 7,5Z"
            />
          </svg>
        </div>
        <input
          name="password"
          onChange={[Function]}
          placeholder="Contraseña..."
          type="password"
        />
        <button
          className="formgroupbutton"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="mdi-icon "
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M12,9A3,3 0 0,0 9,12A3,3 0 0,0 12,15A3,3 0 0,0 15,12A3,3 0 0,0 12,9M12,17A5,5 0 0,1 7,12A5,5 0 0,1 12,7A5,5 0 0,1 17,12A5,5 0 0,1 12,17M12,4.5C7,4.5 2.73,7.61 1,12C2.73,16.39 7,19.5 12,19.5C17,19.5 21.27,16.39 23,12C21.27,7.61 17,4.5 12,4.5Z"
            />
          </svg>
        </button>
      </div>
      <a
        className="css-16ubog-LogInForm"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Olvidaste tu contraseña?
      </a>
    </div>
    <div
      className="MuiButtonGroup-root MuiButtonGroup-fullWidth MuiButtonGroup-vertical"
      role="group"
      style={
        Object {
          "margin": "10px 0",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlined MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Iniciar Sesión
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
